WalletWasabi.Fluent/ViewModels/AddWalletPageViewModel.cs

```diff
@@ -53,6 +58,34 @@ public AddWalletPageViewModel(IScreen screen, WalletManager walletManager, Bitco
 			PasswordInteraction = new Interaction<string, string?>();
 			PasswordInteraction.RegisterHandler(
 				async interaction => interaction.SetOutput(await new EnterPasswordViewModel().ShowDialogAsync()));
+
+			this.ValidateProperty(x => x.WalletName, errors => ValidateWalletName(errors, walletManager, WalletName));
+		}
+
+		private void ValidateWalletName(IValidationErrors errors, WalletManager walletManager, string walletName)
+		{
+			string walletFilePath = Path.Combine(walletManager.WalletDirectories.WalletsDir, $"{walletName}.json");
+
+			if (string.IsNullOrEmpty(walletName))
+			{
+				return;
+			}
+
+			if (File.Exists(walletFilePath))
+			{
+				errors.Add(ErrorSeverity.Error, $"A wallet named {WalletName} already exists. Please try a different name.");
+			}
+
+			if (string.IsNullOrWhiteSpace(walletName))
+			{
+				errors.Add(ErrorSeverity.Error, "Wallet name is not valid.");
+				return;
+			}
+
+			if (!WalletGenerator.ValidateWalletName(walletName))
+			{
+				errors.Add(ErrorSeverity.Error, $"{walletName} is not a valid Wallet name. Please try a different name.");
```

Comment:
What if we just say something like `Selected wallet name is not valid. Please try a different name.`
Or handle this in TextBox's DataValidationErrors?

Inserting tons of space cause this look:
![image](https://user-images.githubusercontent.com/16364053/98036082-b2efae80-1e19-11eb-859e-c271a093fa64.png)